### PR TITLE
Download from s3 with version

### PR DESCRIPTION
**Pull request by KorhanOzturk90**

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/frontendGlobal.scala [hunk 1]
```diff
@@ -125,8 +125,8 @@ class ApplicationModule(context: Context) extends BuiltInComponentsFromContext(c
 
   // quarantine
   lazy val quarantineRepository = quarantine.Repository(db)
-  lazy val retrieveFile = quarantineRepository.retrieveFile _
-  lazy val getFileFromQuarantine = QuarantineService.getFileFromQuarantine(retrieveFile) _
+  lazy val retrieveFileFromQuarantineBucket = s3Service.retrieveFileFromQuarantine _
+  lazy val getFileFromQuarantine = QuarantineService.getFileFromQuarantine(retrieveFileFromQuarantineBucket) _
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/frontendGlobal.scala]: Here you should be able to pass `s3Service.retrieveFileFromQuarantine` directly to `getFileFromQuarantine` without a need for intermediate variable `retrieveFileFromQuarantineBucket`

File: app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala [hunk 2]
```diff
@@ -112,6 +118,18 @@ class S3JavaSdkService extends S3Service {
         s3Client.getObject(new GetObjectRequest(bucketName, key, versionId)).getObjectContent
       }
 
+
+  override def retrieveFileFromQuarantine(key: String, versionId: String)(implicit ec: ExecutionContext) = {
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala]: getting a file from S3 is blocking, we need to ensure that we are using a thread pool, in this case we end up using `play.api.libs.concurrent.Execution.Implicits._` and we shouldn't

Comment (KorhanOzturk90) [inline app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala]: good shout & done

File: app/uk/gov/hmrc/fileupload/frontendGlobal.scala [hunk 3]
```diff
@@ -125,8 +126,10 @@ class ApplicationModule(context: Context) extends BuiltInComponentsFromContext(c
 
   // quarantine
   lazy val quarantineRepository = quarantine.Repository(db)
-  lazy val retrieveFile = quarantineRepository.retrieveFile _
-  lazy val getFileFromQuarantine = QuarantineService.getFileFromQuarantine(retrieveFile) _
+  //lazy val retrieveFileFromQuarantineBucket = s3Service.retrieveFileFromQuarantine _
+  //TODO discuss alternative thread pools
+  lazy val ec = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(25))
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/frontendGlobal.scala]: Not better to take these values to config?

Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/frontendGlobal.scala]: We will need to make a decision on thread pools for all blocking calls. We will discuss a bit later

**Pull request merged**